lib/dao.js

```diff
@@ -2350,6 +2350,7 @@ DataAccessObject.prototype.updateAttributes = function updateAttributes(data, op
               typedData[key] = typedData[key].toObject();
             }
           }
+          data = typedData;
```

Comment:
This is IMO rather evil. Please rework the code below to use `typedData` and `context.data` as we discussed earlier and keep the `data` variable unmodified.